src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -454,6 +454,13 @@ protected function getFormActions()
 
             $gridState = $this->gridField->getState(false);
             $actions->push(HiddenField::create($manager->getStateKey($this->gridField), null, $gridState));
+            if (ClassInfo::hasMethod($manager, 'getStateRequestVar')) {
+                $stateRequestVar = $manager->getStateRequestVar();
+                $stateValue = $this->getRequest()->requestVar($stateRequestVar);
+                if ($stateValue) {
+                    $actions->push(HiddenField::create($stateRequestVar, '', $stateValue));
+                }
+            }
```

Comment:
As per https://github.com/silverstripe/silverstripe-framework/pull/11288#discussion_r1657116190

```suggestion
            if (ClassInfo::hasMethod($manager, 'getStateRequestVar')) {
                $stateRequestVar = $manager->getStateRequestVar();
                $stateValue = $this->getRequest()->requestVar($stateRequestVar);
                if ($stateValue) {
                    $actions->push(HiddenField::create($stateRequestVar, '', $stateValue));
                }
            } else {
                $actions->push(HiddenField::create($manager->getStateKey($this->gridField), null, $gridState));
            }
```

Reply:
This is now changed